server/src/main/java/com/genymobile/scrcpy/CameraEncoder.java

```diff
@@ -0,0 +1,229 @@
+package com.genymobile.scrcpy;
+
+import android.annotation.SuppressLint;
+import android.graphics.Rect;
+import android.hardware.camera2.CameraAccessException;
+import android.hardware.camera2.CameraCaptureSession;
+import android.hardware.camera2.CameraCharacteristics;
+import android.hardware.camera2.CameraDevice;
+import android.hardware.camera2.CaptureRequest;
+import android.hardware.camera2.params.StreamConfigurationMap;
+import android.media.MediaCodec;
+import android.os.Build;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.SystemClock;
+import android.view.Surface;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.Semaphore;
+
+public class CameraEncoder extends SurfaceEncoder {
+
+    private int maxSize;
+    private String cameraId;
+    private CameraPosition cameraPosition;
+
+    private String actualCameraId;
+    private CameraDevice cameraDevice;
+
+    private HandlerThread cameraThread;
+    private Handler cameraHandler;
+
+    public CameraEncoder(int maxSize, String cameraId, CameraPosition cameraPosition, Streamer streamer,
+            int videoBitRate, int maxFps, List<CodecOption> codecOptions, String encoderName, boolean downsizeOnError) {
+        super(streamer, videoBitRate, maxFps, codecOptions, encoderName, downsizeOnError);
+
+        this.maxSize = maxSize;
+        this.cameraId = cameraId;
+        this.cameraPosition = cameraPosition;
+
+        cameraThread = new HandlerThread("camera");
+        cameraThread.start();
+        cameraHandler = new Handler(cameraThread.getLooper());
+    }
+
+    @SuppressLint("MissingPermission")
+    private CameraDevice openCamera(String id)
+            throws CameraAccessException, InterruptedException {
+        Semaphore semaphore = new Semaphore(0);

```

Comment:
I don't know how to properly do this. Maybe with `ExecutorService` and `Future`s?